modules/lang-mustache/src/main/java/org/elasticsearch/action/search/template/TransportMultiSearchTemplateAction.java

```diff
@@ -0,0 +1,78 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.search.template;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.search.MultiSearchResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.AtomicArray;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TransportMultiSearchTemplateAction extends HandledTransportAction<MultiSearchTemplateRequest, MultiSearchResponse> {
+
+    private final TransportSearchTemplateAction searchTemplateAction;
+
+    @Inject
+    public TransportMultiSearchTemplateAction(Settings settings, ThreadPool threadPool, TransportService transportService,
+                                              ActionFilters actionFilters, IndexNameExpressionResolver resolver,
+                                              TransportSearchTemplateAction searchTemplateAction) {
+        super(settings, MultiSearchTemplateAction.NAME, threadPool, transportService, actionFilters, resolver,
+                MultiSearchTemplateRequest::new);
+        this.searchTemplateAction = searchTemplateAction;
+    }
+
+    @Override
+    protected void doExecute(MultiSearchTemplateRequest request, ActionListener<MultiSearchResponse> listener) {
+        final AtomicArray<MultiSearchResponse.Item> responses = new AtomicArray<>(request.requests().size());
+        final AtomicInteger counter = new AtomicInteger(responses.length());
+
+        for (int i = 0; i < responses.length(); i++) {
+            final int index = i;
+            searchTemplateAction.execute(request.requests().get(i), new ActionListener<SearchTemplateResponse>() {
+                @Override
+                public void onResponse(SearchTemplateResponse searchTemplateResponse) {
+                    responses.set(index, new MultiSearchResponse.Item(searchTemplateResponse.getResponse(), null));
```

Comment:
I think we also need a `MultiSearchTemplateResponse` class? Otherwise if we simulated responses are omitted?